controllers/keystoneapi_controller.go

```diff
@@ -819,22 +828,23 @@ func (r *KeystoneAPIReconciler) ensureFernetKeys(
 // createHashOfInputHashes - creates a hash of hashes which gets added to the resources which requires a restart
 // if any of the input resources change, like configs, passwords, ...
 //
+// returns the hash, whether the hash changed (as a bool) and any error
+//
 func (r *KeystoneAPIReconciler) createHashOfInputHashes(
 	ctx context.Context,
 	instance *keystonev1.KeystoneAPI,
 	envVars map[string]env.Setter,
-) (string, error) {
+) (string, bool, error) {
+	var hashMap map[string]string
+	changed := false
 	mergedMapVars := env.MergeEnvs([]corev1.EnvVar{}, envVars)
 	hash, err := util.ObjectHash(mergedMapVars)
 	if err != nil {
-		return hash, err
+		return hash, changed, err
 	}
-	if hashMap, changed := util.SetHash(instance.Status.Hash, common.InputHashName, hash); changed {
+	if hashMap, changed = util.SetHash(instance.Status.Hash, common.InputHashName, hash); changed {
 		instance.Status.Hash = hashMap
-		if err := r.Client.Status().Update(ctx, instance); err != nil {
-			return hash, err
-		}
 		r.Log.Info(fmt.Sprintf("Input maps hash %s - %s", common.InputHashName, hash))
```

Comment:
ditto, w/o updaiting it right away, this log message is too early. It could be rephrased like queued update, or something

Reply:
Well, in this case the message is merely displaying the hash as opposed to including a verb like `added` to indicate that something might have been persisted.  But we could qualify the message with "(queued for update)".

I'm thinking we need some sort of the logging in the upcoming standardized `defer`-ed instance update func ([example](https://github.com/openstack-k8s-operators/lib-common/compare/master...abays:lib-common:helper_instance_update)) to indicate that the instance (and/or its status) was successfully patched/updated at the end of the reconcile loop.